### ISSUE TITLE
[Bug Report] [Component] [date-picker] FocusTrap breaks nested dialog "focus on tab"

### ISSUE THREAD

**Issue by purepear** [element-plus/element-plus#7842]
Bug Type: **`Component`**

## Environment

- Vue Version: `3.2.33`
- Element Plus Version: `2.2.1`
- Browser / OS: `Chrome/MacOS`
- Build Tool: `Vite`

### Related Component

- `el-date-picker`

### Reproduction Link

Select a date by selecting the year from the year table, month from the month table, day from the day table:
[Element Plus Playground](https://element-plus.run/#eyJBcHAudnVlIjoiPHRlbXBsYXRlPlxuICA8ZWwtZGF0ZS1waWNrZXIgOm1vZGVsLXZhbHVlPVwiZGF0ZVwiIEB1cGRhdGU6bW9kZWwtdmFsdWU9XCJ1cGRhdGVEYXRlXCIgLz5cblx0PG9sPlxuICAgIDxsaSB2LWZvcj1cIihldmVudCwgaW5kZXgpIGluIGV2ZW50c1wiIDprZXk9XCJpbmRleFwiPjxwcmU+e3sgZXZlbnQgfX08L3ByZT48L2xpPiAgXG4gIDwvb2w+XG48L3RlbXBsYXRlPlxuXG48c2NyaXB0IGxhbmc9XCJ0c1wiIHNldHVwPlxuaW1wb3J0IHsgcmVmIH0gZnJvbSAndnVlJ1xuXG5jb25zdCBldmVudHMgPSByZWYoW10pXG5cbmNvbnN0IGRhdGUgPSByZWYoJycpXG5jb25zdCB1cGRhdGVEYXRlID0gKHZhbHVlKSA9PiB7XG4gIGRhdGUudmFsdWUgPSB2YWx1ZVxuICBldmVudHMudmFsdWUucHVzaCh2YWx1ZSlcbiAgY29uc29sZS5sb2coJ0VtaXR0ZWQ6JywgdmFsdWUpXG59XG48L3NjcmlwdD5cbiIsImltcG9ydF9tYXAuanNvbiI6IntcbiAgXCJpbXBvcnRzXCI6IHt9XG59IiwiX28iOnt9fQ==)

### Steps to reproduce

Select a date by selecting the year from the years table, then the month from the months table, then the day from the days table.

1. Start by selecting the year label at the top. 
  <img width="338" alt="1" src="https://user-images.githubusercontent.com/144010/169684943-8cc312aa-1a00-422d-80b1-71b8adae5842.png">

2. Selecting 2023 will emit next year's current date
<img width="338" alt="2" src="https://user-images.githubusercontent.com/144010/169684948-afe36fbe-e317-4cef-8b69-8c1a45114508.png">

3. Selecting June will emit next year's first of June
<img width="327" alt="3" src="https://user-images.githubusercontent.com/144010/169684949-50160ebc-cc45-489c-a996-73d87036f271.png">

4. Selecting the day will finally emit the correct day. That's the only emit that should happen in this case.
<img width="331" alt="4" src="https://user-images.githubusercontent.com/144010/169684953-1ced9742-96ea-4447-87b5-1d6f0dc5b5ce.png">

### What is Expected?

Only one "update:model-value" event should be emitted after selecting the day.

### What is actually happening?

There are three "update:model-value" events: one after each selection: year, month, day. 

### Additional comments

I believe the regression was introduced in https://github.com/element-plus/element-plus/pull/7598

More specifically: [here](https://github.com/element-plus/element-plus/pull/7598/files#diff-ce64860e98ab8af028e85764b0975f255abd40120c8544de36e9105b18966759R438-R442
) and [here](https://github.com/element-plus/element-plus/pull/7598/files#diff-ce64860e98ab8af028e85764b0975f255abd40120c8544de36e9105b18966759R454-R458
)
Before creating a PR that removes some of the code linked above, @opengraphica could you share some info about the decision to add that code? And also, thank you so much for making these components more accessible! 🙇

**Comment by opengraphica** [comment]
Much of the existing logic in these components relies on the `parsedValue` prop changing, which can only be updated on emit. It was necessary in order to keep the selection updated as the keyboard moves through the dates.

Much more refactoring would be necessary to fix.

**Comment by zhiyuanzmj** [comment]
use `type="daterange"` also trigger twice "update:model-value" event

**Comment by purepear** [comment]
@opengraphica yeah, i agree there's no easy fix but this is quite a big regression IMO. (For example a datepicker filter that has a watcher doing a http request will do unnecessary http calls).
Should we revert https://github.com/element-plus/element-plus/pull/7598 until we figure out a way to address accessibility without breaking functionality?
